torch/sparse/_triton_ops.py

```diff
@@ -82,7 +82,7 @@ def is_compatible_blocksize(b):
 
 
 def make_triton_contiguous(t):
-    if t.stride(-2) > 1 and t.stride(-1) > 1:
+    if (t.stride(-2) > 1 or t.dtype is torch.float32) and t.stride(-1) > 1:
```

Comment:
This is the gist of the PR: apparently, using contiguous input to `triton.language.dot` pays off the overheads from `contiguous()` call when inputs have float32 dtype and relatively large sizes (>1000).